### PR TITLE
58 add export functionality for dmps with missing funder information

### PR DESCRIPTION
**Pull request by lpandath**
#### What does this PR do?
In DmpActionsComponent, the exportDmpTemplate() method now opens a warning dialog and determines whether to export the DMP template or the DMP itself based on the funderSupported property of the project. This logic was also integrated into the getDocument() method in PlansComponent to ensure consistency across the application.

#### Breaking changes
The DmpActionsComponent has been updated with a new exportDmpTemplate() method that opens the export dialog and dispatches the exportDmpTemplate action if the DMP's funder is not supported, or the exportDmp action otherwise. The PlansComponent now uses a similar approach to handle the export of DMP documents, with the new getDocument() method opening the export dialog and dispatching the appropriate action depending on the funder support of the DMP.

### Checks

- [ ] Summary updated
- [ ] Version view updated
- [ ] Documentation added
- [x] Tests added
- [ ] E2e tests created
- [ ] Successfully ran e2e tests before merge

closes GH-

### CODE REVIEW

**Review by ZenoLC — APPROVED** [review]
looks good to me,
i would appreciate a second review since I am still shaky on angular :)

File: libs/damap/src/lib/components/plans/plans.component.ts [hunk 1]
```diff
@@ -50,31 +59,51 @@ export class PlansComponent implements OnInit {
...
+  }
+
+  openExportWarningDialog(funderSupported: boolean, id: number): void {
+    const dialogRef = this.dialog.open(ExportWarningDialogComponent, {});
+    dialogRef.componentInstance.funderSupported = funderSupported;
+
+    dialogRef.beforeClosed().subscribe(result => {
+      if (result === undefined) {
+        return;
+      } else {
```
Comment (rekt-hard) [inline libs/damap/src/lib/components/plans/plans.component.ts:74]: Else can be removed, as we return in the if before. Also reduces complexity when reading code

File: libs/damap/src/lib/components/plans/plans.component.ts [hunk 2]
```diff
@@ -50,31 +59,51 @@ export class PlansComponent implements OnInit {
...
+        const template = result;
```
Comment (rekt-hard) [inline libs/damap/src/lib/components/plans/plans.component.ts:75]: variable can be removed and assigned directly two lines after that.

File: libs/damap/src/lib/domain/enum/export-template-type.enum.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export enum ETemplateType {
+    FWF="FWF",
+    SCIENCE_EUROPE="SCIENCE_EUROPE"
+}
```
Comment (rekt-hard) [inline libs/damap/src/lib/domain/enum/export-template-type.enum.ts:4]: new line missing

File: libs/damap/src/lib/components/dmp/dmp-actions/dmp-actions.component.spec.ts [hunk 4]
```diff
@@ -99,17 +102,48 @@ describe('DmpActionsComponent', () => {
...
+    let dialogs = await loader.getAllHarnesses(MatDialogHarness);
+    expect(dialogs.length).toBe(1);
+
+    await dialogs[0].close();
+    dialogs = await loader.getAllHarnesses(MatDialogHarness);
+
+    expect(dialogs.length).toBe(0);
+    expect(component.exportDmpTemplate).toHaveBeenCalledTimes(1);
+    expect(component.dispatchExportDmpTemplate).toHaveBeenCalledTimes(1);
   });
```
Comment (rekt-hard) [inline libs/damap/src/lib/components/dmp/dmp-actions/dmp-actions.component.spec.ts:148]: These tests might not cover the logic, which has to be tested. Whether `dispatchExportDmp` or `dispatchExportDmpTemplate` is called, depends on the project. The respective function should also be called automatically before the dialog is closed and not manually (as in the testcase).

File: libs/damap/src/lib/widgets/export-warning-dialog/export-warning-dialog.component.spec.ts
Comment (rekt-hard) [inline libs/damap/src/lib/widgets/export-warning-dialog/export-warning-dialog.component.spec.ts:1]: Why are these changes required?

**Review by rekt-hard — COMMENTED** [review]
Functionality is there.
Tests might need some more work, imo.
Any possibility to make the `exportDmp` methods reusable or put them in a `utils` file? They are quite similar with regards to what they do and it would ease maintenance of the method.

File: libs/damap/src/lib/widgets/export-warning-dialog/export-warning-dialog.component.spec.ts
Comment (rekt-hard) [reply libs/damap/src/lib/widgets/export-warning-dialog/export-warning-dialog.component.spec.ts:1]: all the changes in the file. I do not see any reason for these changes in the tests.

**Review by rekt-hard — APPROVED** [review]
Another test case for the plans component would increase coverage. Other than that, looking good :)
